### PR TITLE
backend/fix: #1123 Added new  parameter, Bonus to driver/profile/stat…

### PR DESCRIPTION
**Pull request by ratnadeep-99**
…s API.

## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 1]
```diff
@@ -873,16 +879,30 @@ respondQuote (driverId, _) req = do
...
   return $
     DriverStatsRes
       { totalRidesOfDay = length rides,
-        totalEarningsOfDay = sum . catMaybes $ rides <&> (.fare)
+        totalEarningsOfDay = sum . catMaybes $ rides <&> (.fare),
+        bonusEarning =
+          let (driverSelFares, customerExtFees) = (catMaybes $ fareParameters <&> (.driverSelectedFare), catMaybes $ fareParameters <&> (.customerExtraFee))
+              driverSelFares' = getMoney <$> driverSelFares
+              customerExtFees' = getMoney <$> customerExtFees
+           in Money $ L.sum driverSelFares' + L.sum customerExtFees' + length driverSelFares' * 10
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:905]: sum fareParameters.fareParametersDetails.deadKmFare 
not the length

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 2]
```diff
@@ -75,6 +75,7 @@ import qualified Domain.Types.Vehicle as Veh
 import qualified Domain.Types.Vehicle.Variant as Variant
 import Environment
 import EulerHS.Prelude hiding (id, state)
+import qualified GHC.List as L
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:78]: GHCL

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 3]
```diff
@@ -873,16 +879,30 @@ respondQuote (driverId, _) req = do
...
   (Id SP.Person, Id DM.Merchant) ->
   Day ->
   m DriverStatsRes
 getStats (driverId, _) date = do
   rides <- runInReplica $ QRide.getRidesForDate driverId date
+  fareParameters' <-
+    mapM
+      ( \ride ->
+          case ride.fareParametersId of
+            Just id -> QFP.findById id
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:892]: runInReplica $  QFP.findById id

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 4]
```diff
@@ -873,16 +879,30 @@ respondQuote (driverId, _) req = do
         Notify.notifyDriverClearedFare orgId driverReq.driverId driverReq.searchTryId driverQuote.estimatedFare driver_.deviceToken
 
 getStats ::
-  (EsqDBReplicaFlow m r, EncFlow m r) =>
+  (EsqDBReplicaFlow m r, EncFlow m r, HasField "esqDBEnv" r EsqDBEnv) =>
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:882]: remove ` HasField "esqDBEnv" r EsqDBEnv)`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 5]
```diff
@@ -86,7 +87,10 @@ import qualified Kernel.External.SMS.MyValueFirst.Types as SMS
 import Kernel.Prelude (NominalDiffTime)
 import Kernel.Sms.Config
 import qualified Kernel.Storage.Esqueleto as Esq
-import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
+import Kernel.Storage.Esqueleto.Config
+  ( EsqDBEnv,
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:91]: not needed

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs (hunk 3 above)
Comment (hkmangla) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:892]: `fareParameters' <- mapM (maybe (pure Nothing) QFP.findById . (.fareParametersId)) rides`

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs [hunk 6]
```diff
@@ -879,10 +883,26 @@ getStats ::
   m DriverStatsRes
 getStats (driverId, _) date = do
   rides <- runInReplica $ QRide.getRidesForDate driverId date
+  let fareParamId = mapMaybe (.fareParametersId) rides
+  fareParameters <- catMaybes <$> mapM (runInReplica . QFP.findById) fareParamId
```
Comment (piyushKumar-1) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs:887]: hey, this can be a in query ?

File: Backend/.envrc
Comment (srid) [inline Backend/.envrc:1]: `.envrc` is gitignored.

**Pull request closed without merging**